src/scripts/icon-to-rn.ts

```diff
@@ -28,43 +10,89 @@ if (!fs.existsSync(outputDirectory)) {
 }
 
 const iconsJsonContent = fs.readFileSync(iconsJsonPath, 'utf8');
-const iconsData: IconType = JSON.parse(iconsJsonContent);
+const iconsData: IconaDataType = JSON.parse(iconsJsonContent);
 
-const transformedIcons: {
-  [name: string]: {
-    solidPath: string;
-    outlinePath: string;
-  };
-} = {};
+const transformedIcons: TransformedIconsType = {};
 
 Object.values(iconsData).forEach(icon => {
   const iconNames = icon.name.split('_');
-  const iconType = iconNames.pop();
+  const iconType = iconNames.pop() ?? "default";
   const iconName = iconNames.join('-');
-  const pathType = iconType === 'yes' ? 'solidPath' : 'outlinePath';
   const filename = path.basename(toPascalCase(iconName.split('/')[1]), '.tsx');
-  const svgContent = icon.svg;
-  const svgRegex = /<svg[^>]*>([\s\S]*?)<\/svg>/;
-  const svgMatch = svgContent.match(svgRegex);
+  const svgMatch = getSvgInnerHTML(icon.svg);
   if (svgMatch) {
-    const svgPath = removeFillAttribute(svgMatch[1]);
-    transformedIcons[filename] = { ...transformedIcons[filename], [pathType]: svgPath };
+    const svgPath = removeFillAttribute(svgMatch);
+    transformedIcons[filename] = { ...transformedIcons[filename], [iconType]: svgPath };
   }
 });
 
 Object.entries(transformedIcons).forEach(([name, paths]) => {
   const outputFilePath = path.join(outputDirectory, `${name}Icon.tsx`);
+  const header = `/**\n * ${new Date()}에 자동 생성됨\n */`;
+  const stringifiedPaths = Object.entries(paths).map(([type, path]) => {
+    return `\n\t"${type}": '${path.replace(/\n/g, '')}'`;
+  })
+
   try {
-    const outputFileContent = `import { iconGenerator } from '../utils/icon-utils';
+    const outputFileContent = `${header}
+import Variables from './DEPRECATED_Variables';
+import { colorValues } from '../design-tokens/main/react-native';
 
-const ${name}Icon = iconGenerator({
-  solidPath: \`
-    ${paths.solidPath.trim()}
-  \`,
-  outlinePath: \`
-    ${paths.outlinePath.trim()}
-  \`
-});
+type TokenColorsType = keyof typeof colorValues;
+type VariablesColorsType = keyof typeof Variables.themeColors;
+type IconColorType = VariablesColorsType | TokenColorsType;
+
+const paths = {${stringifiedPaths.join(',')}
+};
+
+const isTokenColorType = (color: any): color is TokenColorsType => {
+  return color in colorValues;
+}
+
+const getIconColors = (fill?: IconColorType, stroke?: IconColorType) => {
```

Comment:
```suggestion
const getIconColors = ({fill, stroke}: {fill?: IconColorType; stroke?: IconColorType;}) => {
```

optional 파라미터가 섞여 있는 경우는 가능하면 object 형태였으면 합니다. 좀 헷갈리더라고요 호출할 때